api/configuration/v1/configsource.go

```diff
@@ -11,6 +12,7 @@ type ConfigSource struct {
 // It is an equivalent of the following patch:
 // `{"op": "add", "path": {.Path}, "value": {.ComputedValueFrom}}`.
 // +kubebuilder:object:generate=true
+// +kgosupported
 type ConfigPatch struct {
```

Comment:
I wonder if those types here are really considered supported by KGO 🤔.

Those are used in https://github.com/Kong/kubernetes-configuration/blob/d4d87e1acd2d5a7d333c30504c1d0aa97e6b9809/api/configuration/v1/kongplugin_types.go#L73 which is included in `kongPlugin` which is used by KGO but fields like `ConfigPatch` and `ConfigSource` are not supported in KGO.

IMHO those should not land in KGO's documentation. This can be a way of trying to hide some of the fields that may be technically available for users in KGO (because we can't add CEL rules on CRDs which are also used by KIC) but should not be set because they will have not effect.

LMKWYT

cc: @mheap

Reply:
That's a good point. I removed the marker from the types. I also had to add another marker to selectively exclude fields from the supported types (like the mentioned `ConfigPatches` and `ConfigSource` fields in `KongPlugin`).

Reply:
👍 

Can we add a comment somewhere about how to use these markers? It can be a net new README section. WDYT?